### PR TITLE
Start pds automatically

### PR DESCRIPTION
**Pull request by eagraf**
* Refactors node and hdb startup a bit. Instead of passing an initial state, an initial list of transitions is passed in.
* Also pins a toolchain version because that was causing issues

### CODE REVIEW

**Comment by sashankg** [comment]
Didn't see this before opening https://github.com/eagraf/habitat-new/pull/17 but you wanna land #17 separately and rebase.

File: internal/node/controller/init.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+			Version: "1",
+			UserID:  constants.RootUserID,
+			Package: node.Package{
+				Driver: "docker",
+				DriverConfig: map[string]interface{}{
+					"env": []string{
+						"PDS_HOSTNAME=ethangraf.com",
+						"PDS_DATA_DIRECTORY=/pds",
+						"PDS_BLOBSTORE_DISK_LOCATION=/pds/blocks",
+						"PDS_PLC_ROTATION_KEY_K256_PRIVATE_KEY_HEX=5290bb1866a03fb23b09a6ffd64d21f6a4ebf624eaa301930eeb81740699239c",
```
Comment (arushibandi) [inline internal/node/controller/init.go:33]: hmm should this key be in here?

**Review by arushibandi — COMMENTED** [review]
one question

File: internal/node/controller/init.go (hunk 1 above)
Comment (eagraf) [reply internal/node/controller/init.go:33]: Yeah I don't want to spend too much time about getting the configuration of PDS perfect right now.

**Review by arushibandi — APPROVED** [review]

**Pull request merged**